Decorator/GuzzleClientFactoryDecorator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auxmoney\OpentracingGuzzleBundle\Decorator;
+
+
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleRequestSpanning;
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleTracingHeaderInjection;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class GuzzleClientFactoryDecorator
+{
+
+    private $decorated;
+
+    public function __call($method, $args)
+    {
+        if (is_callable($this->decorated, $method)) {
+
+            /** @var Client $client */
+            $client = call_user_func_array(array($this->decorated, $method), $args);
+
+            $handler = $client->getConfig("handler");
+            if (is_object($handler) && is_a($handler, HandlerStack::class)) {
+                $handler->push(new Reference(GuzzleRequestSpanning::class));
+                $handler->push(new Reference(GuzzleTracingHeaderInjection::class));
+            }
+
+            return $client;
+
+        }
+        throw new Exception(
```

Comment:
This is still WIP? Because right now, it will always throw this exception.

Reply:
@cawolf yes it's an early wip to design the bones of the feature, thanks for your review :)